File: csharp/2014/teenage-mutant-ninja-turtles-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Teenage Mutant Ninja Turtles" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Teenage Mutant Ninja Turtles",
                year: 2014,
                plot: "When a kingpin threatens New York City, a group of mutated turtle warriors must emerge from the shadows to protect their home.",
                rating: 5.9
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Teenage Mutant Ninja Turtles",
                year: 2014
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2014/teenage-mutant-ninja-turtles-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Teenage Mutant Ninja Turtles to the database
movies.insert(
    title="Teenage Mutant Ninja Turtles", 
    year=2014, 
    plot="When a kingpin threatens New York City, a group of mutated turtle warriors must emerge from the shadows to protect their home.", 
    rating=5.9
)

# Confirm that the movie was added
movie = movies.select(
    title="Teenage Mutant Ninja Turtles", 
    year=2014
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    